### PR TITLE
add --from-backup flag to ark restore create & allow restore naming

### LINKED ISSUE #60: Consideration for refactoring restore command
When creating a restore, I find it a bit confusing that the `backup`'s name is in the form of an argument, as we aren't actually creating a `backup`, it is just a piece of a given restore.  

One option would be: 
- Have the backup be an explicitly named, required flag.  
- Optionally support an argument which would then be the restore's name itself, not that of the backup

Current: `ark restore create <backup name>`
Proposed: `ark restore create --backup <backup name>`

A broader question would be, has there been consideration to flipping the verb and resource to more closely match how Kubectl works?  `ark create restore ...`

### PR DESCRIPTION
**Pull request by skriss**
Signed-off-by: Steve Kriss <steve@heptio.com>

Fixes #60

### CODE REVIEW

File: pkg/cmd/cli/restore/create.go [hunk 1]
```diff
@@ -38,7 +38,7 @@ func NewCreateCommand(f client.Factory, use string) *cobra.Command {
 	o := NewCreateOptions()
 
 	c := &cobra.Command{
-		Use:   use + " BACKUP",
+		Use:   use + " [NAME] --from-backup <backup-name>",
```
Comment (skriss) [inline pkg/cmd/cli/restore/create.go]: Not sure this is entirely clear, open to input on how best to document that restore name is optional, --from-backup is required

**Comment by nrb** [comment]
Nitpick - could you fix up the commit message so it's not wrapping?

**Comment by skriss** [comment]
done

File: pkg/cmd/cli/restore/create.go (hunk 1 above)
Comment (jrnt30) [reply pkg/cmd/cli/restore/create.go]: For the `Use` I think this is the form I would expect.  If you wanted to elaborate on it, I think the `Short`, `Long` or `Examples` would be the best place to add that additional context

**Comment by jrnt30** [comment]
Haven't looked into the new PR process with docs, but should there be an associated doc update for the examples in the PR that introduces the changes?

File: pkg/cmd/cli/restore/create.go (hunk 1 above)
Comment (ncdc) [reply pkg/cmd/cli/restore/create.go]: I would put `--from-backup` first, e.g. `--from-backup BACKUP_NAME [RESTORE_NAME]`. Putting something in `[braces]` typically denotes that it's optional. And +1 to Justin's suggestion of using short/long/example to show what happens both with and without the restore name.

File: pkg/cmd/cli/restore/create.go [hunk 2]
```diff
@@ -97,8 +99,12 @@ func (o *CreateOptions) BindFlags(flags *pflag.FlagSet) {
 }
 
 func (o *CreateOptions) Validate(c *cobra.Command, args []string, f client.Factory) error {
-	if len(args) != 1 {
-		return errors.New("you must specify only one argument, the backup's name")
+	if len(o.BackupName) == 0 {
+		return errors.New("you must specify a backup to restore from, via --from-backup")
```
Comment (ncdc) [inline pkg/cmd/cli/restore/create.go]: Could make this shorter: `--from-backup is required`

**Comment by skriss** [comment]
addressed all feedback, PTAL.

**Comment by lypht** [comment]
I think this a great addition; my full support.

**Comment by ncdc** [comment]
@skriss let's proceed. Please squash into either 2 or 1 commit.

**Comment by skriss** [comment]
squashed.

**Comment by skriss** [comment]
switched order of restore name and backup name in docs/examples.

**Comment by ncdc** [comment]
LGTM

**Pull request merged**